HHS/Head-Start-TTADP
# [TTAHUB-1360] Don't save on form navigation if the form isn't dirty

## Description of change

A prop `saveWhenClean` was added that, when false, and the form is not dirty, the form does not get saved. The result is really snappy behavior when using the navigator to move around a clean form.

https://user-images.githubusercontent.com/17074357/225965495-4e47b2db-8276-4551-b40a-426f20117276.mov


## How to test

* The tests should still pass.
* Navigate around an AR form and notice the lack of saving on a clean form.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1360


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
